### PR TITLE
Rework player implementation setting

### PR DESCRIPTION
**Pull request by Maxr1998**
Allow to switch between web player, ExoPlayer and external player via client settings

### CODE REVIEW

File: app/src/main/java/org/jellyfin/mobile/settings/VideoPlayerType.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+@StringDef({
+        VideoPlayerType.WEB_PLAYER,
+        VideoPlayerType.EXO_PLAYER,
+        VideoPlayerType.EXTERNAL_PLAYER
+})
+public @interface VideoPlayerType {
+    String WEB_PLAYER = "web";
+    String EXO_PLAYER = "exo";
+    String EXTERNAL_PLAYER = "external";
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/mobile/settings/VideoPlayerType.java:14]:
```suggestion
    String WEB_PLAYER = "webui";
    String EXO_PLAYER = "exoplayer";
    String EXTERNAL_PLAYER = "external";
```

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]
Looks good, some notes though:

- The "allow playing audio in the background..." setting should be disabled when the video player is NOT native.
- The "external video player" isn't vlc or mxplayer only but those contain some extra code to work better because android apparently doesn't provide a useful api for that

**Comment by Maxr1998** [comment]
> * The "allow playing audio in the background..." setting should be disabled when the video player is NOT native.

For that, I'd have to do further changes to the settings library first (as in, add a callback). Don't know if it's important enough to do that now or keep that for later. The preference at least says it's for the native player only.

**Comment by nielsvanvelzen** [comment]
We can do it later but we should make an issue then so we won't forget it.

**Review by nielsvanvelzen — APPROVED** [review]

**Pull request merged**